object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -386,9 +390,13 @@ private void alwaysCallTraverse(ControlFlowGraph cfg) {
 
           if (receiver instanceof LocalVariableNode
               && isVarInDefs(newDefs, (LocalVariableNode) receiver)) {
-            if (getAlwaysCallValue(((LocalVariableNode) receiver).getElement())
-                .equals(method.getSimpleName().toString())) {
-              // If the method called on the receiver is the same as receiver's @AlwaysCall value,
+            List<String> mustCallVal = getMustCallValue(((LocalVariableNode) receiver).getTree());
+            // TODO: I think this will cause false positives if there is more than one value in the
+            // @MustCall annotation
+
+            if (mustCallVal.size() == 1
+                && mustCallVal.get(0).equals(method.getSimpleName().toString())) {
```

Comment:
I think we should remove the assignment tree from the newDefs if mustCallVal.size() is greater than 1 and the node is the last required method that needs to be called. But, it's not important for this pull request I can add this check later.